### PR TITLE
Run sample validation in CI/CD and validate that starting page works

### PR DESCRIPTION
**Pull request by asilverman**
# Description

Adds automatic validation to the main tutorial. This approach leverages our functional test framework to execute the tutorial manifests and assert on their deployment. There are some gaps identified for which issues were filed and TODO comments were added to reflect those gaps.

This PR introduces a new requirement to the execution context. To run the samples tests the execution context must set `PROJECT_RADIUS_SAMPLES_REPO_ABS_PATH` to the absolute path of the project-radius/samples repository, otherwise the tests will be skipped.

## Issue reference
Fixes https://github.com/project-radius/radius/issues/4349

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Adds necessary unit tests for change
* [x] Adds necessary E2E tests for change
* [x] Unit tests passing
* [x] Extended the documentation / Created issue for it

### CODE REVIEW

**Review by willdavsmith — APPROVED** [review]

**Review by jkotalik — APPROVED** [review]
Let's get another approver before merging as I co-authored some of this PR

**Review by willdavsmith — APPROVED** [review]

**Review by jkotalik — APPROVED** [review]

**Pull request merged**
